odoo/addons/base/models/res_currency.py

```diff
@@ -356,6 +358,9 @@ def create(self, vals_list):
         return super().create([self._sanitize_vals(vals) for vals in vals_list])
 
     def _get_latest_rate(self):
+        # Make sure 'name' is defined when creating a new rate.
+        if not self.name:
+            raise UserError(_("The date for the current rate is empty.\nPlease set it."))
```

Comment:
I'm not sure about the decorator arguments of `_onchange_rate_warning` though

Reply:
What do you mean, would you add another field besides `company_rate` ?

Reply:
I would expect `rate` and `name` but not `company_rate`, don't you?